orca-peering/src/main/kotlin/com/netflix/spinnaker/orca/peering/ExecutionCopier.kt

```diff
@@ -152,13 +154,35 @@ open class ExecutionCopier(
         destDB.loadRecords(getStagesTable(executionType).name, stageRows)
       }
 
+      val partitionField = DSL.field("partition")
+
       // Step 2: Copy all executions
-      executionRows.forEach { r -> r.set(DSL.field("partition"), peeredId)
-        latestUpdatedAt = max(latestUpdatedAt, r.get("updated_at", Long::class.java))
+      if (executionRows.size < idsToMigrate.size) {
+        // There must've been some executions that have been deleted during our diff calculation, find out which ones
+        val idsToMigrateHash = idsToMigrate.toHashSet()
+
+        executionRows.forEach { r ->
+          r.set(partitionField, peeredId)
+          latestUpdatedAt = max(latestUpdatedAt, r.get("updated_at", Long::class.java))
+          idsToMigrateHash.remove(r.get("id"))
+        }
+
+        executionsNoLongerPresent.addAll(idsToMigrateHash)
+      } else {
+        executionRows.forEach { r ->
+          r.set(partitionField, peeredId)
+          latestUpdatedAt = max(latestUpdatedAt, r.get("updated_at", Long::class.java))
+        }
```

Comment:
The forEach introduces quite a bit of duplication in the 2 branches of the if/else, wouldn't it be the same as this?

```suggestion
      executionRows.forEach { r ->
        r.set(partitionField, peeredId)
        latestUpdatedAt = max(latestUpdatedAt, r.get("updated_at", Long::class.java))
      }
      
      if (executionRows.size < idsToMigrate.size) {
        // There must've been some executions that have been deleted during our diff calculation, find out which ones
        val idsToMigrateHash = idsToMigrate.toHashSet()

        executionRows.forEach { r ->
          idsToMigrateHash.remove(r.get("id"))
        }

        executionsNoLongerPresent.addAll(idsToMigrateHash)
      }```

Reply:
I believe this is maybe a slightly more efficient/readable way to figure out executions no longer present:

```
  private fun executionsNoLongerPresent(executionRows: Result<Record>, idsToMigrate: List<String>): Iterable<String> {
    return if (executionRows.size < idsToMigrate.size) {
      // There must've been some executions that have been deleted during our diff calculation, find out which ones
      val executionSet = executionRows.intoMap(DSL.field("id"))
      idsToMigrate.filter { id -> !executionSet.containsKey(id) }
    } else {
      emptyList()
    }
  }
```

and then it's a simple call

```
if (executionsNoLongerPresent(executionRows, idsToMigrate).any()) {
   ...
```